### PR TITLE
RIA-9214 add_fee_service_to_case_api

### PR DESCRIPTION
**Pull request by ikirsanov**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RIA-9214

### Change description ###
Updated ManageFeeUpdateHandler(when the hearing type is changed the costs changed as well). Added FeesHelper services. Updated AddedValidationPreparer with new conditions.

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by jamieclarkeHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java [hunk 1]
```diff
@@ -1,59 +1,62 @@
...
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.PaymentStatus;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class AddedValidationPreparer implements PreSubmitCallbackHandler<AsylumCase> {
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java:20]: @ikirsanov, are you using a different formatting? Why is the whole file has diff?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesHelper.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import java.math.BigDecimal;
+import java.util.Optional;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.fee.Fee;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.fee.FeeType;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.FeeService;
+
+public class FeesHelper {
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesHelper.java:22]: Can we add unit test for this class?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesAndStatusCheckPreparerTest.java [hunk 3]
```diff
@@ -1,344 +1,345 @@
...
-            feesAndStatusCheckPreparer.handle(PreSubmitCallbackStage.ABOUT_TO_START, callback);
-
-        assertNotNull(callbackResponse);
-        assertThat(callbackResponse.getErrors()).isNotEmpty();
-        assertThat(callbackResponse.getErrors())
-            .contains("The Make a payment option is not available.");
-    }
-
-}
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit.payment;
```
Comment (ThomasKKC) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesAndStatusCheckPreparerTest.java:1]: Can you fix the formatting here too? It's hard to review when the whole file has diff

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java [hunk 4]
```diff
@@ -1,59 +1,62 @@
...
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        final PreSubmitCallbackResponse<AsylumCase> asylumCasePreSubmitCallbackResponse = new PreSubmitCallbackResponse<>(asylumCase);
+
+        if (asylumCase.read(PAYMENT_STATUS, PaymentStatus.class).equals(Optional.of(PaymentStatus.PAID))
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java:55]: Do you need to update the FT 'RIA-3409-error-on-payment-when-appeal-in-paidStatus'?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/FeesRegisterApi.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.clients;
+
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.fee.FeeResponse;
+
+@FeignClient(name = "fees-register-api", url = "${fees-register.api.url}")
+public interface FeesRegisterApi {
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/FeesRegisterApi.java:10]: We need to add contract testing for this

**Review by ThomasKKC — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesHelper.java (hunk 2 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesHelper.java:22]: Added the test

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesAndStatusCheckPreparerTest.java (hunk 3 above)
Comment (ikirsanov) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/FeesAndStatusCheckPreparerTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java (hunk 4 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java:55]: done

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java (hunk 1 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/AddedValidationPreparer.java]: Done

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/FeesRegisterApi.java (hunk 5 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/FeesRegisterApi.java:10]: added

**Review by ThomasKKC — APPROVED** [review]

**Pull request merged**